cmd/ddev/cmd/flags_test.go

```diff
@@ -0,0 +1,193 @@
+	assert := asrt.New(t)
+
+	var subject Flags
+	var c cobra.Command
+	var f *pflag.Flag
+
+	// No flags
+	subject = getSubject()
+	c = getCommand()
+	assert.NoError(subject.AssignToCommand(&c))
+
+	// Minimal
+	assert.NoError(subject.LoadFromJSON(`[{"Name":"test","Usage":"Usage of test"}]`))
+	assert.NoError(subject.AssignToCommand(&c))
+
+	f = c.Flags().Lookup("test")
+	assert.NotEmpty(f)
+	assert.EqualValues("test", f.Name)
+	assert.EqualValues("", f.Shorthand)
+	assert.EqualValues("Usage of test", f.Usage)
+	assert.EqualValues(FtBool, f.Value.Type())
+	assert.EqualValues("false", f.DefValue)
+	assert.EqualValues("", f.NoOptDefVal)
+	assert.Empty(f.Annotations)
+
+	// Full
+	subject = getSubject()
+	c = getCommand()
+	assert.NoError(subject.LoadFromJSON(`[{"Name":"test-1","Shorthand":"t","Usage":"Usage of test 1","Type":"bool","DefValue":"true","NoOptDefVal":"true","Annotations":{"test-1":["test-1-1","test-1-2"]}},{"Name":"test-2","Usage":"Usage of test 2","Type":"string","DefValue":"DefValue","NoOptDefVal":"NoOptDefVal"},{"Name":"test-3","Usage":"Usage of test 3","Type":"int","DefValue":"1","NoOptDefVal":"-1"},{"Name":"test-4","Usage":"Usage of test 4","Type":"uint","DefValue":"1","NoOptDefVal":"2"}]`))
+	assert.NoError(subject.AssignToCommand(&c))
+
+	f = c.Flags().Lookup("test-1")
+	assert.NotEmpty(f)
+	assert.EqualValues("test-1", f.Name)
+	assert.EqualValues("t", f.Shorthand)
+	assert.EqualValues("Usage of test 1", f.Usage)
+	assert.EqualValues(FtBool, f.Value.Type())
+	assert.EqualValues("true", f.DefValue)
+	assert.EqualValues("true", f.NoOptDefVal)
+	assert.EqualValues(map[string][]string{"test-1": {"test-1-1", "test-1-2"}}, f.Annotations)
+
+	f = c.Flags().Lookup("test-2")
+	assert.EqualValues("test-2", f.Name)
+	assert.EqualValues("", f.Shorthand)
+	assert.EqualValues("Usage of test 2", f.Usage)
+	assert.EqualValues(FtString, f.Value.Type())
+	assert.EqualValues("DefValue", f.DefValue)
+	assert.EqualValues("NoOptDefVal", f.NoOptDefVal)
+	assert.Empty(f.Annotations)
+
+	f = c.Flags().Lookup("test-3")
+	assert.EqualValues("test-3", f.Name)
+	assert.EqualValues("", f.Shorthand)
+	assert.EqualValues("Usage of test 3", f.Usage)
+	assert.EqualValues(FtInt, f.Value.Type())
+	assert.EqualValues("1", f.DefValue)
+	assert.EqualValues("-1", f.NoOptDefVal)
+	assert.Empty(f.Annotations)
+
+	f = c.Flags().Lookup("test-4")
+	assert.EqualValues("test-4", f.Name)
+	assert.EqualValues("", f.Shorthand)
+	assert.EqualValues("Usage of test 4", f.Usage)
+	assert.EqualValues(FtUint, f.Value.Type())
+	assert.EqualValues("1", f.DefValue)
+	assert.EqualValues("2", f.NoOptDefVal)
+	assert.Empty(f.Annotations)
+
+	// No fully implemented type
+	subject = getSubject()
+	c = getCommand()
+	assert.NoError(subject.LoadFromJSON(`[{"Name":"test","Usage":"Usage of test","Type":"_test3_"}]`))
+	assert.PanicsWithValue("Error implementation missing for type '_test3_'", func() {
+		assert.NoError(subject.AssignToCommand(&c))
```

Comment:
As I'm checking here for a panic this check makes imho no sense but `make staticrequired` complains if this is missing. Same above btw.

Reply:
Wow, I've never seen an assert for a panic! 

You can use `//nolint: <linter>` to specify the lint-checking behavior you want, for example `// nolint: errcheck`. 

It's not clear to me what makes no sense (the func()?) but anyway, you can turn off linting for a specific linter.

Reply:
It makes no sense to check the return value of the method because it does not return anything but should panic instead.

Yeah, checking for a panic is ofc an edge case. But here I like to make sure the method panics to be sure dev errors are discovered during the CI :)

But no need to change the linter behavior here but nice to know there are ways to fix such cases for the future. Thanks for that hint!

Reply:
Ah, if it's errcheck that's the issue, just use `_ = whateverfunction();` and all the linters will be quiet. Easier than `//nolint errcheck`. "Throw away the return value. Yes I know I'm doing it."